src/popup/components/Modals/TransferReceive.vue

```diff
@@ -101,95 +115,102 @@ export default {
-    } else {
-      this.selectedAsset = this.getAeternityToken({
-        tokenBalance: this.balance,
-        balanceCurrency: this.balanceCurrency,
-      });
-    }
-  },
-  methods: {
-    getTokenInfo(link = false) {
-      if (this.amount <= 0) return '';
-      const token = this.selectedAsset && (this.selectedAsset.symbol === 'AE'
+  setup(props, { root }) {
+    const amount = ref<string | number>(0);
+    const selectedAsset = ref<IAsset | IToken | null>(null);
+
+    const balance = rxJsObservableToVueState(root.$store.state.observables.balance);
+    const balanceCurrency = rxJsObservableToVueState(root.$store.state.observables.balanceCurrency);
+
+    const account = computed<IAccount>(() => root.$store.getters.account);
+    const availableTokens = computed<ITokenList>(
+      () => root.$store.state.fungibleTokens.availableTokens,
+    );
+    const getAeternityToken = computed(() => root.$store.getters['fungibleTokens/getAeternityToken']);
+
+    const { copied, copy } = useCopy();
+
+    function getTokenInfo(link = false) {
+      if (amount.value <= 0) return '';
+      const token = selectedAsset && (selectedAsset.value?.symbol === 'AE'
         ? 'AE'
-        : this.selectedAsset.contractId);
+        : selectedAsset.value?.contractId);
       const firstChar = link ? '&' : '?';
       const separator = link ? '&' : '&amp;';
-      return `${firstChar}token=${token}${separator}amount=${this.amount}`;
-    },
-    getLink(value) {
-      return `${APP_LINK_WEB}/account?account=${value}${this.getTokenInfo(true)}`;
-    },
-    async share() {
-      const { address } = this.account;
-      const walletLink = this.getLink(address);
-      const text = (this.amount > 0)
-        ? this.$t('modals.receive.shareTextNoAmount', { address, walletLink })
-        : this.$t('modals.receive.shareTextWithAmount', { address, walletLink, amount: this.amount });
-      await this.$store.dispatch('share', { text });
-    },
-    getQRdata(address) {
-      return this.amount > 0
-        ? `${this.getLink(address)}`
-        : address;
-    },
-    getTextToCopy() {
-      return this.amount > 0
-        ? this.getLink(this.account.address)
-        : this.account.address;
-    },
-    handleAssetChange(asset) {
-      this.selectedAsset = asset;
-    },
-    closeModal() {
-      this.$store.commit('modals/closeByKey', MODAL_TRANSFER_RECEIVE);
-    },
+      return `${firstChar}token=${token}${separator}amount=${amount.value}`;
+    }
+
+    function getLink(value: string) {
+      return `${APP_LINK_WEB}/account?account=${value}${getTokenInfo(true)}`;
+    }
+
+    const accountAddressAsLink = computed(() => {
+      return (amount.value > 0)
+        ? getLink(account.value.address)
+        : account.value.address;
+    });
+
+    const computedAddress = computed(() => {
+      return (amount.value > 0)
+        ? `${account.value.address}?${getTokenInfo(true).substring(1)}`
+        : account.value.address;
+    });
```

Comment:
we could remove `return` in both functions

Reply:
Sorry, but:
![image](https://user-images.githubusercontent.com/8217670/194861976-8ab3e368-3f08-47f7-aaac-d4f73ee8dc3d.png)
https://eslint.org/docs/latest/rules/no-confusing-arrow